### DISCUSSION TITLE
Provide a way to disable ExUnit integration for production use

### DISCUSSION THREAD

**Discussion by lackac** [keathley/wallaby#589] [Ideas]
I'm using Wallaby in a production environment to automate filling of a 3rd party web form. It's not ideal, but we needed a solution even while we're working with the 3rd party on an API based integration.

We are deploying to production with releases built by `mix release` and this doesn't include `ExUnit` by default which we would like to keep that way. However Wallaby is trying to ensure that the `:ex_unit` application is started and attempts to create an `ExUnit.after_suite/1` hook in `Wallaby.SessionStore`. This fails currently in a release because `ExUnit` is not bundled in it.

I didn't see a way to disable ExUnit integration but I think that would be a useful option for similar use cases. There are several ways this could be achieved:

1. a configuration option that explicitly disables ExUnit integration entirely
2. implicitly disable ExUnit integration in `Wallaby.SessionStore` by making it conditional bases on `function_exported?(ExUnit, :after_suite, 1)`
3. provide a way for an alternative implementation of `Wallaby.SessionStore`

I think option 2 would make the most sense in the current state of the project. Option 3 would be overkill for just this use case while the configuration based approach of option 1 is likely unnecessary for that single place of integration that is causing the issue. It's unlikely anyone would use `Wallaby.Feature` in a production setting where most of the ExUnit integration exists.

**Comment by lambdachad** [comment]
Hey did you find a solution for this?

**Comment by mhanberg** [comment]
You might want to check out https://github.com/elixir-wallaby/web_driver_client instead. 

Wallaby is a testing library, which uses web_driver_client in some spaces. 

You can use web_driver_client without any test modules getting in the way.

**Comment by lambdachad** [comment]
I forked the repository and added check which fixes this issue. My fork is here: https://github.com/svelterust/wallaby

The change consists of adding if-statement around the ExUnit loading code if `:ex_unit` exists: https://github.com/elixir-wallaby/wallaby/commit/19768af7d94ca485a65efc0dba9560bb83e20dc8
